joomla/joomla-cms
# [7.0] Add offline mode support in the API

Pull Request for Issue # .

### Summary of Changes
Adds support for the `onBeforeRender` and `onAfterRender` events to the API. It should be noted our core plugin correctly checks for html document types or if in site/admin modes in most places. The only exception to this is the cache plugin. However I think this won't change much because we don't call `onAfterRoute` in the API which does all the initialisation anyhow.

Adds support for offline mode to the API - if the site is in offline mode the API now should not be made available to users who cannot access it alongside an appropriate 503 status code

### Testing Instructions
Turn on Offline mode. Before this PR check that the API is accessible. After PR you get a 503 error code.

For the Plugin events this should be as easy as dropping some test functions into an existing plugin.

### Documentation Changes Required
N/A
